### PR TITLE
Read object error checking

### PR DESCRIPTION
**Pull request by dangitall**

### CODE REVIEW

File: src/act.wizard.c [hunk 1]
```diff
@@ -2135,6 +2135,10 @@ void do_load(struct char_data *ch, char *argument,
       return;
     }
     mob = read_mobile(number, REAL);
+    if (!mob) {
+      send_to_char("Sorry, error loading mosnter. File a bug.\n\r", ch);
```
Comment (jonm) [inline src/act.wizard.c]: Typo mosnter -> monster

File: src/act.wizard.c [hunk 2]
```diff
@@ -2632,16 +2640,39 @@ void do_start(struct char_data *ch) {
...
+    }
+    else {
+      obj = read_object(r_num, REAL);
+    }
+    if (!obj) {
+      send_to_char("Game error loading object. File a bug.\n\r", ch);
+    }
+    else {
+      obj_to_char(obj, ch);       /* bread   */
+    }
```
Comment (jonm) [inline src/act.wizard.c]: This seems a little awkward, putting the `!obj` test in front of every line. What about extracting this function instead?

```
void load_obj_for_char(int obj_vnum, struct char_data *ch) {
  struct obj_data *obj;
  int r_num = real_object(obj_vnum);
  if (r_num < 0) {
    send_to_char("Game error loading object. File a bug.\n\r", ch);
    return;
  }
  obj = read_object(r_num, REAL);
  if (!obj) {
    send_to_char("Game error loading object. File a bug.\n\r", ch);
    return;
  }
  obj_to_char(obj, ch);
}
```

and then you can just assign the breads with:

```
load_obj_for_char(12, ch);
load_obj_for_char(12, ch);
```

Or, perhaps `load_obj_for_char` could take a quantity too.

File: src/db.c [hunk 3]
```diff
@@ -1240,29 +1243,40 @@ struct char_data *read_mobile(int nr, int type) {
   mob->player.title = 0;
 
   /* *** Numeric data *** */
+#ifdef VERIFY_COUNT
+  #undef VERIFY_COUNT
+#endif
+#define VERIFY_COUNT(got, expect)                       \
+  do {                                                  \
+    if ((got) < expect) {                               \
```
Comment (jonm) [inline src/db.c]: Should have parens around `expect` too?

Comment (dangitall) [inline src/db.c]: Sure.

**Comment by jonm** [comment]
Gotta run someone to swim team practice, will resume review upon return. :)

File: src/act.wizard.c (hunk 2 above)
Comment (dangitall) [inline src/act.wizard.c]: I agree, that is better.

**Comment by jonm** [comment]
Ok, last comment would be: instead of (or perhaps in addition to) telling the player there was an error and they should file a bug, perhaps use `log_msgf` too?

**Comment by dangitall** [comment]
Do you mean for the NULL checks? I avoided this mostly because log messages will be generated internally by the  `read_mobile` and `read_object` functions themselves with more precise info (the fp position). But if you think it would be more useful to have context (or there is some other place that an error slipped through unlogged), lmk!

**Comment by jonm** [comment]
Ah, ok, no if a log message gets generated elsewhere, then I am :+1:​

**Pull request merged**